pkg/operator/ceph/cluster/osd/config.go

```diff
@@ -31,11 +31,24 @@ key = %s
 `
 )
 
-func (c *Cluster) generateKeyring(osdID string, d *apps.Deployment) error {
-	resourceName := d.GetName()
-
+func (c *Cluster) generateKeyring(osdID string, deploymentName string) error {
```

Comment:
I think this is the comment you mentioned maybe having some questions about? When Rook creates the deployment, it gives the deployment an owner reference to `c.ownerRef`, which is a reference to the `CephCluster` which controls the deployment IIRC. When initially creating the keyring, I think it would be good to give it this same owner reference so that the keyring secret is deleted at the same time as the `CephCluster` in the event that a vestigial secret is left behind due to some series of unfortunate events which results in keyring secret creation but where the secret can't later be set as a child of the deployment.